src/core/layertree/qgslayertreemodel.cpp

```diff
@@ -941,6 +941,14 @@ void QgsLayerTreeModel::connectToLayer( QgsLayerTreeLayer *nodeLayer )
     connect( vl, &QgsVectorLayer::layerModified, this, &QgsLayerTreeModel::layerNeedsUpdate, Qt::UniqueConnection );
   }
 
+  if ( layer->type() == QgsMapLayerType::MeshLayer )
```

Comment:
Same here -- now that the signals are in the QgsMapLayer class I'd create the connections for ALL layers, regardless of their type.